src/views/catalogi/CatalogiDetails.vue

```diff
@@ -178,65 +178,57 @@ export default {
 			handler(newCatalogiItem, oldCatalogiItem) {
 				if (!this.upToDate || JSON.stringify(newCatalogiItem) !== JSON.stringify(oldCatalogiItem)) {
 					this.catalogi = newCatalogiItem
-					newCatalogiItem && this.fetchData(newCatalogiItem?.id)
+					if (newCatalogiItem) {
+						this.loading = true
+						catalogiStore.getCatalog(newCatalogiItem.id)
+							.then(() => {
+								this.catalogi = catalogiStore.catalogiItem
+								this.loading = false
+							})
+					}
 					this.upToDate = true
-					newCatalogiItem?.organization ? this.fetchOrganization(newCatalogiItem?.organization) : this.organization = false
+					if (newCatalogiItem?.organization) {
+						this.organizationLoading = true
+						organizationStore.getOrganization(newCatalogiItem.organization)
+							.then(() => {
+								this.organization = organizationStore.organizationItem
+								this.organizationLoading = false
+							})
+					} else {
+						this.organization = false
+					}
 				}
 			},
 			deep: true,
 		},
 	},
 	mounted() {
 		this.catalogi = catalogiStore.catalogiItem
-		// check if catalogiItem is not false
-		catalogiStore.catalogiItem && this.fetchData(catalogiStore.catalogiItem?.id)
-
-		catalogiStore.catalogiItem.organization && this.fetchOrganization(catalogiStore.catalogiItem.organization)
-
-		this.publicationTypeLoading = true
-	},
-	methods: {
-		fetchData(catalog) {
+		if (catalogiStore.catalogiItem) {
 			this.loading = true
-			fetch(
-				`/index.php/apps/opencatalogi/api/catalogi/${catalog}`,
-				{
-					method: 'GET',
-				},
-			)
-				.then((response) => {
-					response.json().then((data) => {
-						catalogiStore.setCatalogiItem(data)
-						this.catalogi = catalogiStore.catalogiItem
-						publicationTypeStore.refreshPublicationTypeList()
-							.then(() => {
-								this.publicationTypeLoading = false
-							})
-					})
+			catalogiStore.getCatalog(catalogiStore.catalogiItem.id)
+				.then(() => {
+					this.catalogi = catalogiStore.catalogiItem
 					this.loading = false
 				})
-				.catch((err) => {
-					console.error(err)
-					this.loading = false
+		}
+
+		if (catalogiStore.catalogiItem.organization) {
+			this.organizationLoading = true
+			organizationStore.getOrganization(catalogiStore.catalogiItem.organization)
```

Comment:
`getOrganization` bestaat niet, gebruik `getOneOrganization`
```suggestion
			organizationStore.getOneOrganization(catalogiStore.catalogiItem.organization)
```

Reply:
@SudoThijn zou jij deze kunnen oppakken?